ledger/complete/wal/checkpointer.go

```diff
@@ -314,34 +320,57 @@ func StoreCheckpoint(writer io.Writer, tries ...*trie.MTrie) error {
 		return fmt.Errorf("cannot write checkpoint header: %w", err)
 	}
 
-	// Nodes are serialized in two steps to save memory.
```

Comment:
looks like most of the changes to this file is actually v5 specific? maybe break this into a different PR

Reply:
That's right. 

I forgot to mention, this PR is [on top of this PR actually](https://github.com/onflow/flow-go/pull/3139/files), which includes these v5 specific changes. Better to review that first.